src/main/java/com/internet/shop/dao/ShoppingCartDao.java

```diff
@@ -0,0 +1,18 @@
+package com.internet.shop.dao;
+
+import com.internet.shop.model.Product;
+import com.internet.shop.model.ShoppingCart;
+
+public interface ShoppingCartDao {
+    ShoppingCart create(ShoppingCart shoppingCart);
+
+    ShoppingCart addProduct(ShoppingCart shoppingCart, Product product);
```

Comment:
it's enough to have update method on dao layer instead of addProduct + deleteProduct, because they represent different changes to products field of your model.